src/cargo/util/toml/mod.rs

```diff
@@ -3146,6 +3157,120 @@ impl<'de> de::Deserialize<'de> for TomlDebugInfo {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Ord, PartialOrd, Hash, Serialize)]
+#[serde(untagged)]
+pub enum TomlTrimPaths {
+    One(TomlTrimPathsValue),
+    Many(Vec<TomlTrimPathsValue>),
+}
+
+impl<'de> de::Deserialize<'de> for TomlTrimPaths {
+    fn deserialize<D>(d: D) -> Result<TomlTrimPaths, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        let expecting = r#"a boolean, "none", "macro", "diagnostics", "object", "all", or a list of these options"#;
+        UntaggedEnumVisitor::new()
+            .expecting(expecting)
+            .bool(|value| {
+                Ok(if value {
+                    TomlTrimPaths::all()
+                } else {
+                    TomlTrimPaths::none()
+                })
+            })
+            .string(|value| {
+                TomlTrimPathsValue::deserialize(value.into_deserializer()).map(TomlTrimPaths::One)
+            })
+            .seq(|seq| {
+                let seq: Vec<String> = seq.deserialize()?;
+                let seq = seq
+                    .into_iter()
+                    .map(|s| TomlTrimPathsValue::deserialize(s.into_deserializer()))
+                    .collect::<Result<_, _>>()?;
+                Ok(TomlTrimPaths::Many(seq))
```

Comment:
Can we not deserliaize directly into `Vec<TomlTrimPathsValue>`?

Reply:
Could you say more on the modeling you're looking for, or is there any existing code I can take as a reference?

Reply:
Can't you do
```rust
let seq: Vec<TomlTrimPathsValue> = seq.deserialize()?;
```
Whether you deserialzie the `Vec` to have `String` or `TomlTrimPathsValue` shouldn't make a difference to serde.

Reply:
```
invalid type: string "diagnostics", expected enum TomlTrimPathsValue
```

That was what I got with `let seq: Vec<TomlTrimPathsValue> = seq.deserialize()?;`. Not yet figure out why. A bit odd.

Reply:
Baffled by this but not a reason to block being merged